packages/office-addin-usage-data/src/usageData.ts

```diff
@@ -282,12 +284,11 @@ export class OfficeAddinUsageData {
    */
   public maskFilePaths(err: Error): Error {
     try {
-      const regexRemoveUserFilePaths = /[\/\\](.*)[\/\\]/gim; /* eslint-disable-line no-useless-escape */
-      const regexRemoveAbsoluteUserFilePathsFromStack = /\w:\\(?:[^\\\s]+\\)+/gim;
-      const regexRemoveFirstFilePathFromStack = /\\(.*)\./i;
-      err.message = err.message.replace(regexRemoveUserFilePaths, "\\");
-      err.stack = err.stack.replace(regexRemoveFirstFilePathFromStack, "\\.");
-      err.stack = err.stack.replace(regexRemoveAbsoluteUserFilePathsFromStack, "");
+      const regexRemoveUserFilePaths = /\w:[\/\\](?:[^\\\s]+[\/\\])+/gmi;
+
+      err.message = err.message.replace(regexRemoveUserFilePaths, "");
+      err.stack = err.stack.replace(regexRemoveUserFilePaths, "");
+
```

Comment:
Why all these changes?  We shouldn't change anything here.

Reply:
The code was wrong. It simply didn't mask the files sometimes.
The tests were passing because they were relying on the order they were executed.

When I made the mask function not have side effects for example, the tests started not passing.

Reply:
But now we're leaving it to have the "side effects" so this doesn't need to change (the less you change the less chance you have of unintentionally breaking something).  Aside from that I can't tell if the new expression covers all the cases the 3 previous expressions did without doing too much.

Reply:
My bad for not communicating this. I had a plan for these functions to not have side effects, but then you talked about it being fine (which I agree with for the `maskFilePaths`). So what I did was to remove the side effects from the `sendUsageDataException` and the `reportExpectedException`  which had them (and called `maskFilePaths`), and I also removed the side effects from the `test.ts`, which had multiple tests using the same variable (the `err` global variable).   

So side effects from the public functions and tests were removed, but the side effect from `maskFilePaths` was kept.  

About the fact that you cannot tell if the expression covers all the cases that were previously covered. The tests are still passing, the same tests we had.

Reply:
Please ensure this meets compliance requirements.

Reply:
@akrantz It should fix the error from before (which could not parse some paths) and that's it. No other behaviors are expected.

Reply:
I'm not certain we should be removing all file names.  You made that change before to address the first file in the stack specifically and I'm thinking that's because the rest of the stack mostly includes information from libraries that would be useful in figuring out what is going on.  If we're going to change these regexs at all we should be very targeted to what's needed for compliance.  We want to remove information considered user data, but not so much information that the remaining data becomes useless (what's the point in keeping the drive letter and the extension).

Reply:
Maintaining the drive letter and extensions is very useless I agree. It just made it easier to write the regexes, as it is very hard for these two to contain users' information.

We can't just remove the first file from the stack, because other files from it might contain user's info.
I think there's still point in keeping the message and logging it, even without the files on it.

If we know the message, we can later do a search on the code and find where that message is. We them can see what the error is about if a lot of users are having the problem, so it is not useless.

Reply:
The message is certainly useful and I'm not questioning having that.  

I was thinking that call stacks are useful and are limited when we don't know the files from the libraries we provide that are part of the stack.  One could argue that the file names are going to be public when the package is created anyway (which is the reason we have file names from libraries that are not theirs and not ours) since it's all ts/js.

However, I suppose to be on the safe side we shouldn't have the file paths at all (even the drive letter and extension).  If we're going to change this now we should change all file paths to just be a token (like "<FilePath>") indicating it was a file path in the message or the stack.

Reply:
Yeah, it is more to be on the safe side. I just added the `""` string for us to know this was filtered from the original message.
I also made the regex remove the drive letter and extension as well

Reply:
The blank "" doesn't tell us what was removed.  It could be that we mask other things per requirements and then we can't tell which is which . . . thus the suggestion of some sort of token.